### PR TITLE
Add banner to conversations

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
The commits d4522f8846ac...17f0200911b5 introduce visual changes on linux. <details><summary>:mag_right: 3 new, 1 changed</summary>

 * Added: **Chat_Banner-Invite**  
   Chat_Banner-Invite rendered
 * Added: **Chat_Banner-Info**  
   Chat_Banner-Info rendered
 * Added: **Chat_Banner-Error**  
   Chat_Banner-Error rendered
 * Changed: **Paper_Key-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480362352730-d4522f8846ac-17f0200911b5-linux/Paper_Key-Normal-d4522f8846ac.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480362352730-d4522f8846ac-17f0200911b5-linux/Paper_Key-Normal-17f0200911b5.png)  
   Paper_Key-Normal comparison
</summary>

File: shared/chat/conversation/banner.desktop.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import React from 'react'
+import {Box, Icon, Text, Header as CommonHeader} from '../../common-adapters'
+import {globalStyles, globalColors, globalMargins} from '../../styles'
+
+import type {Props, ErrorVariant, InviteVariant} from './banner'
+
+const ErrorBanner = (props: ErrorVariant) => {
+  return (
+    <CommonHeader style={{...globalStyles.flexBoxColumn, backgroundColor: globalColors.red}} type='Strong'>
+      <Text type='Header' backgroundMode='Announcements' style={{flex: 1, ...globalStyles.flexBoxCenter, paddingTop: 6, cursor: 'default'}}>{props.text}</Text>
```
Comment (chrisnojima) [inline shared/chat/conversation/banner.desktop.js]: why do we need the cursor set here?

File: shared/chat/conversation/banner.desktop.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+import React from 'react'
+import {Box, Icon, Text, Header as CommonHeader} from '../../common-adapters'
+import {globalStyles, globalColors, globalMargins} from '../../styles'
+
+import type {Props, ErrorVariant, InviteVariant} from './banner'
+
+const ErrorBanner = (props: ErrorVariant) => {
+  return (
+    <CommonHeader style={{...globalStyles.flexBoxColumn, backgroundColor: globalColors.red}} type='Strong'>
```
Comment (chrisnojima) [inline shared/chat/conversation/banner.desktop.js]: We should likely remove the windowdragging stuff from commonheader and make that a prop

File: shared/chat/conversation/index.desktop.js [hunk 3]
```diff
@@ -5,15 +5,23 @@ import {globalStyles} from '../../styles'
...
-)
+const Conversation = (props: Props) => {
+  const bannerMessage: ?BannerMessage = props.bannerMessage
+  // $FlowIssue with variants
+  const banner = bannerMessage && <Banner {...bannerMessage} />
+  return (
+    <Box style={{...globalStyles.flexBoxColumn, flex: 1}}>
+      <Header {...props} />
+      <List {...props} />
+      {props.bannerMessage && banner}
```
Comment (chrisnojima) [inline shared/chat/conversation/index.desktop.js]: isn't this just {banner}, its checked above?

File: shared/chat/conversation/banner.desktop.js (hunk 1 above)
Comment (MarcoPolo) [inline shared/chat/conversation/banner.desktop.js]: We don't, probably copy/paste error

**Comment by keybase-ci-visdiff** [comment]
The commits d4522f8846ac...91f20182404d introduce visual changes on linux. <details><summary>:mag_right: 3 new</summary>

 * Added: **Chat_Banner-Invite**  
   Chat_Banner-Invite rendered
 * Added: **Chat_Banner-Info**  
   Chat_Banner-Info rendered
 * Added: **Chat_Banner-Error**  
   Chat_Banner-Error rendered
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 46c73e86b64c...ac795eb5db80 introduce visual changes on linux. <details><summary>:mag_right: 3 new</summary>

 * Added: **Chat_Banner-Invite**  
   Chat_Banner-Invite rendered
 * Added: **Chat_Banner-Info**  
   Chat_Banner-Info rendered
 * Added: **Chat_Banner-Error**  
   Chat_Banner-Error rendered
</summary>

**Pull request merged**